includes/functions/functions_email.php

```diff
@@ -64,32 +57,32 @@
     function zen_mail($to_name, $to_address, $email_subject, $email_text, $from_email_name, $from_email_address, $block = [], $module = 'default', $attachments_list = '', $email_reply_to_name = '', $email_reply_to_address = '')
     {
         global $db, $messageStack, $zco_notifier;
-        if (SEND_EMAILS != 'true') {
+        if (SEND_EMAILS !== 'true') {
             return false;
         }  // if sending email is disabled in Admin, just exit
 
-        if (defined('DEVELOPER_OVERRIDE_EMAIL_STATUS') && DEVELOPER_OVERRIDE_EMAIL_STATUS == 'false') {
+        if (defined('DEVELOPER_OVERRIDE_EMAIL_STATUS') && DEVELOPER_OVERRIDE_EMAIL_STATUS === 'false') {
             return false;
         }  // disable email sending when in developer mode
-        if (defined('DEVELOPER_OVERRIDE_EMAIL_ADDRESS') && DEVELOPER_OVERRIDE_EMAIL_ADDRESS != '') {
+        if (defined('DEVELOPER_OVERRIDE_EMAIL_ADDRESS') && DEVELOPER_OVERRIDE_EMAIL_ADDRESS !== '') {
             $to_address = DEVELOPER_OVERRIDE_EMAIL_ADDRESS;
         }
 
         // ignore sending emails for any of the following pages
         // (The EMAIL_MODULES_TO_SKIP constant can be defined in a new file in the "extra_configures" folder)
-        if (defined('EMAIL_MODULES_TO_SKIP') && in_array($module, explode(",", constant('EMAIL_MODULES_TO_SKIP')))) {
+        if (defined('EMAIL_MODULES_TO_SKIP') && in_array($module, explode(',', constant('EMAIL_MODULES_TO_SKIP')))) {
             return false;
         }
 
         // check for injection attempts. If new-line characters found in header fields, simply fail to send the message
         foreach ([$from_email_address, $to_address, $from_email_name, $to_name, $email_subject] as $key => $value) {
-            if (preg_match("/\r/i", $value) || preg_match("/\n/i", $value)) {
+            if (strpos($value, "\r") !== false || strpos($value, "\n") !== false) {
                 return false;
             }
         }
 
         // if no text or html-msg supplied, exit
-        if (trim($email_text) == '' && (empty($block) || (isset($block['EMAIL_MESSAGE_HTML']) && $block['EMAIL_MESSAGE_HTML'] == ''))) {
+        if (trim($email_text) === '' && empty($block['EMAIL_MESSAGE_HTML'])) {
```

Comment:
Should this still check for `empty($block)`?

Reply:
I thought long and hard on that and I came to the conclusion that if the EMAIL_MESSAGE_HTML element isn't `empty` that that implies that `$block` isn't empty either and, since that array element is present, that `$block` is also an array.

Reply:
hmmm, I think I see your point

It's still a (tiny) functional difference ... BUT then again why would one pass OTHER $block elements but NOT EMAIL_MESSAGE_HTML .. AND leave $email_text empty. 
ie: why send an html-only content message using only uncommon $block array elements?
I suppose a customized email message could use that, but then they should probably also provide text-only content themselves.
So, i guess that's a long way of thinking-out and saying I think i agree with your change! 👍

Reply:
Agreed on the teeny functional difference, but I've been trying to get these highly-used functions and classes to be a tad more performant and that additional `empty` check just felt like overkill.

I appreciate you taking the time to review!